puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/foreman-deb.groovy

```diff
@@ -10,100 +10,12 @@ pipeline {
-                                set_job_build_description("${env.cico_job_name}")
-                            }
-                            deleteDir()
-                        }
-                    }
-                }
-
-                stage('ubuntu1604') {
-                    agent { label 'el' }
-                    environment {
-                        cico_job_name = "foreman-nightly-ubuntu1604-test"
-                    }
-
-                    steps {
-                        git_clone_foreman_infra()
-
-                        withCredentials([string(credentialsId: 'centos-jenkins', variable: 'PASSWORD')]) {
-                            runPlaybook(
-                                playbook: 'ci/centos.org/ansible/jenkins_job.yml',
-                                extraVars: [
-                                    "jenkins_job_name": "${env.cico_job_name}",
-                                    "jenkins_username": "foreman",
-                                    "jenkins_job_link_file": "${env.WORKSPACE}/jobs/${env.cico_job_name}"
-                                ],
-                                sensitiveExtraVars: ["jenkins_password": "${env.PASSWORD}"]
-                            )
-                        }
-                    }
-                    post {
-                        always {
-                            script {
-                                set_job_build_description("${env.cico_job_name}")
-                            }
-                            deleteDir()
-                        }
-                    }
-                }
-
-                stage('ubuntu1804') {
-                    agent { label 'el' }
-                    environment {
-                        cico_job_name = "foreman-nightly-ubuntu1804-test"
-                    }
-
-                    steps {
-                        git_clone_foreman_infra()
-
-                        withCredentials([string(credentialsId: 'centos-jenkins', variable: 'PASSWORD')]) {
-                            runPlaybook(
-                                playbook: 'ci/centos.org/ansible/jenkins_job.yml',
-                                extraVars: [
-                                    "jenkins_job_name": "${env.cico_job_name}",
-                                    "jenkins_username": "foreman",
-                                    "jenkins_job_link_file": "${env.WORKSPACE}/jobs/${env.cico_job_name}"
-                                ],
-                                sensitiveExtraVars: ["jenkins_password": "${env.PASSWORD}"]
-                            )
-                        }
-                    }
-                    post {
-                        always {
-                            script {
-                                set_job_build_description("${env.cico_job_name}")
-                            }
-                            deleteDir()
-                        }
-                    }
-                }
+            steps {
```

Comment:
No steps in this case, parallel acts like steps and this should error if you try to run it with both at the top level. However see https://github.com/theforeman/foreman-infra/pull/989/files#r268856449

Reply:
I'm using paralllel() and there it failed on me so I added this again.